api/trace/trace_context_propagator.go

```diff
@@ -29,21 +28,33 @@ const (
 	supportedVersion  = 0
 	maxVersion        = 254
 	traceparentHeader = "traceparent"
+	tracestateHeader  = "tracestate"
+)
+
+type traceContextPropagatorKeyType uint
+
+const (
+	tracestateKey traceContextPropagatorKeyType = 0
 )
 
 // TraceContext propagates SpanContext in W3C TraceContext format.
 //nolint:golint
 type TraceContext struct{}
 
 var _ propagation.HTTPPropagator = TraceContext{}
-var traceCtxRegExp = regexp.MustCompile("^[0-9a-f]{2}-[a-f0-9]{32}-[a-f0-9]{16}-[a-f0-9]{2}-?")
+var traceCtxRegExp = regexp.MustCompile("^([0-9a-f]{2})-([a-f0-9]{32})-([a-f0-9]{16})-([a-f0-9]{2})(?:-.*)?$")
```

Comment:
may be worth commenting here to indicate which field is which in the regex to make it easier to read/review while you're here.

Reply:
Sadly, Go doesn't support the `/x` regexp modifier, so I named the capture groups as a form of documenting intent.